File: app/GuardianAngel/app/src/main/java/com/example/guardianangel/RatingListCustomAdapter.java
package com.example.guardianangel;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.RatingBar;
import android.widget.TextView;

import java.util.ArrayList;

public class RatingListCustomAdapter extends BaseAdapter {

    ArrayList<String> ratingItemName;
    Context context;
    RatingListCustomAdapter(Context context, ArrayList<String> list){
        ratingItemName = list;
        this.context = context;
    }

    @Override
    public int getCount() {
        return 0;
    }

    @Override
    public Object getItem(int i) {
        return null;
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View v = view;
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        v = inflater.inflate(R.layout.rating_bar_list_item, viewGroup, false);


        TextView itemName = (TextView) v.findViewById(R.id.textView);
        itemName.setText(ratingItemName.get(i).toString());
        RatingBar ratingBar = (RatingBar) v.findViewById(R.id.ratingBar);

        ratingBar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                /*...Here you can get which rating bar is clicked.. where i is the position of the rating bar in list */
            }
        });

        ratingBar.setOnRatingBarChangeListener(new RatingBar.OnRatingBarChangeListener() {
            @Override
            public void onRatingChanged(RatingBar ratingBar, float v, boolean b) {
                //Here handle your rating bar clicked operation ... get id of clicked rating bar
                Log.d("rating_bar_clicked","Rating bar id: "+ratingBar.getId()+ " Value:"+String.valueOf(v));
            }
        });


        return v;
    }
}

File: app/GuardianAngel/app/src/main/java/com/example/guardianangel/RatingListCustomAdapter.kt
package com.example.guardianangel;

import android.annotation.SuppressLint
import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.MotionEvent
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ListView
import android.widget.RatingBar;
import android.widget.TextView;
import android.widget.Toast

import java.util.ArrayList;

class RatingListCustomAdapter(private val context: Context, private val dataList: List<String>) : BaseAdapter() {
    var hashMap: HashMap<String, Float> = HashMap<String, Float>()

//    hashMap["Nausea"] = 0.0f
//    hashMap["Headache"] = 0.0f
//    hashMap["Diarrhoea"] = 0.0f

    override fun getCount(): Int {
        return dataList.size
    }

    override fun getItem(position: Int): Any {
        return dataList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }



    @SuppressLint("ClickableViewAccessibility")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view: View
        val viewHolder: ViewHolder

        for (item in dataList) {
            hashMap[item] = 0.0f
        }
        if (convertView == null) {
            // Inflate the layout for each row
            view =
                LayoutInflater.from(context).inflate(R.layout.rating_bar_list_item, parent, false)

            // Initialize ViewHolder
            viewHolder = ViewHolder(view)
            view.tag = viewHolder
        } else {
            // Reuse the recycled view
            view = convertView
            viewHolder = view.tag as ViewHolder
        }

        // Set data to views
        val item = getItem(position) as String
        viewHolder.ratingBar.rating = hashMap[item]!!
        viewHolder.textView.text = item
        viewHolder.ratingBar.setOnRatingBarChangeListener { _, rating, fromUser ->
            if (fromUser) {
                hashMap[item] = rating
                Log.i("daysDiff", hashMap.toString())

            }
        }

        viewHolder.ratingBar.setOnClickListener {}
        return view
    }
    fun getHashMap(): Map<String, Float> {
        return hashMap
    }


    private class ViewHolder(view: View) {
        val textView: TextView = view.findViewById(R.id.textView)
        val ratingBar: RatingBar = view.findViewById(R.id.ratingBar)
    }
}
